### ISSUE TITLE
Comparison versus other solutions

### ISSUE THREAD

**Issue by CaptainZippy** [vslavik/bakefile#56]
Lots of projects probably have discussions like https://github.com/ColinH/PEGTL/pull/7
It would be nice if the differences from other systems were succinctly described. I'm thinking of:

* architectural choices (especially if it forces things in the project layout)
* features (and deliberate non-features!)
* stability

Similar systems which come to mind are:

* https://premake.github.io/
* http://www.cmake.org/

**Comment by kirbyfan64** [comment]
IMO, the main difference is that Bakefile encourages the build files to be pre-generated (and maybe uploaded to version control), while the others prefer the user to generate the file.

**Comment by vadz** [comment]
Yes, bakefile reason for existence is to allow end users to just build the project without installing anything else on their machine and using the "native" toolset that they're used to.
